sympy/tensor/array/arrayop.py

```diff
@@ -208,3 +208,38 @@ def derive_by_array(expr, dx):
             return ImmutableDenseNDimArray([expr.diff(i) for i in dx], dx.shape)
         else:
             return diff(expr, dx)
+
+
+def permutedims(expr, perm):
+    """
+    Permutes the indices of an array.
+
+    Parameter specifies the permutation of the indices.
+    """
+    if not isinstance(expr, NDimArray):
+        raise TypeError("expression has to be an N-dim array")
+
+    perm = tuple(perm)
```

Comment:
There is a more serious problem: _Permutation_ is constructed by the vector of positions after the permutation, while all other conventions are to use the positions from which the elements came.

Constructing `p = Permutation(perm)` gives you the inverse permutation, if we follow the conventions of other libraries.